### PR TITLE
Fix back handler for Android

### PR DESCRIPTION
**Pull request by parasharrajat**
### Fixed Issues

$ https://github.com/Expensify/App/issues/4211

### Tests | QA Steps
1. Open LHN on Android.
2. navigate to a report.
3. Use the back button on the Android device.
4. You should land on LHN.
5. Use the back button on the Android device.
6. App should be closed.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android (affected)

#### Android

https://user-images.githubusercontent.com/24370807/141691522-e19fa22b-62d8-4e96-bbc8-11e78a766fd8.mp4

### CODE REVIEW

**Comment by parasharrajat** [comment]
@Beamanator I have fixed the IOS freezing issues which caused the last PR to be reverted.  All of the changes are same as per last PR and just adjusted the package version for Reanimated so that it does not cause the freeze this time.

Please test it and this should be ready for merge.

**Comment by parasharrajat** [comment]
Wait, I think I need to clean this up.

**Comment by parasharrajat** [comment]
@Beamanator 
Ok, I have done the testing on each platform, and the only issue that is remaining and for which I update the Reanimated package in the previous PR is point 2 on https://github.com/Expensify/App/pull/5745#issuecomment-946200059.

But I found out that this issue already exists in https://github.com/Expensify/App/issues/5591. Just increasing the package version increased its severity. 

On https://github.com/Expensify/App/issues/5591, when we change the orientation, it switches the drawer from side to permanent, and the exact same thing happens on point 2 where we move from Mobile view to web.

but but but I have a solution for that. 

remember in this previous PR we were using the `userLegacyImplementation` prop to remount the drawer when screening switches between mobile and web. We can do the same by using the `key` prop(we don't have userLegacyImplementation) now it fixes that issue. But I will do that separately into different PR as a solution for that issue. **Proposal on that issue as well.** So after that is merged I will proceed with this PR.
image

**Comment by parasharrajat** [comment]
This is ready.
PR for #5591 is #6356.

**Comment by Beamanator** [comment]
Wow nice work here, and thanks for linking to the other PR where you fixed the orientation issue! I'll review this today :)

File: src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js [hunk 1]
```diff
@@ -3,8 +3,8 @@ import _ from 'underscore';
 import PropTypes from 'prop-types';
 import {createDrawerNavigator} from '@react-navigation/drawer';
 import {View} from 'react-native';
-import styles, {getNavigationDrawerStyle, getNavigationDrawerType} from '../../../styles/styles';
 import withWindowDimensions, {windowDimensionsPropTypes} from '../../../components/withWindowDimensions';
+import styles, {getNavigationDrawerStyle, getNavigationDrawerType} from '../../../styles/styles';
```
Comment (Beamanator) [inline src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js:7]: Why move this down here? I kinda like it better above `withWindowDimensions` to make imports slightly more alphabetical

Comment (Beamanator) [reply src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js:7]: Not a big deal, I'd rather get this merged haha

**Review by Beamanator — APPROVED** [review]
Looks awesome, fantastic work here dude 👍 

I also verified it didn't cause either of these regressions that the last one caused
- https://github.com/Expensify/App/issues/6150
- https://github.com/Expensify/App/issues/6147

File: src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js (hunk 1 above)
Comment (parasharrajat) [reply src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js:7]: Oh, what I am 9 minutes late.... :panda_face:

Comment (Beamanator) [reply src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js:7]: lol it's fine - i thought you would be later :D

**Pull request merged**